apps/src/sites/studio/pages/projects/index.js

```diff
@@ -38,54 +45,50 @@ $(document).ready(() => {
       </Provider>,
       publicGallery);
   });
-});
-
-function showGallery(gallery) {
-  $('#angular-my-projects-wrapper').toggle(gallery === Galleries.PRIVATE);
-  $('#public-gallery-wrapper').toggle(gallery === Galleries.PUBLIC);
-}
 
-function onShowConfirmPublishDialog(callback) {
   const publishConfirm = document.getElementById('publish-confirm');
+
   ReactDOM.render(
-    <Dialog
-      title={i18n.publishToPublicGallery()}
-      body={i18n.publishToPublicGalleryWarning()}
-      confirmText={i18n.publish()}
-      isOpen={true}
-      handleClose={hideDialog}
-      onCancel={hideDialog}
-      onConfirm={onConfirmPublish.bind(this, callback)}
-    />,
+    <Provider store={store}>
+      <PublishDialog/>
+    </Provider>,
     publishConfirm
   );
-}
+});
 
-// Make this method available to angularProjects.js. This can go away
-// once My Projects is moved to React.
-window.onShowConfirmPublishDialog = onShowConfirmPublishDialog;
+function showGallery(gallery) {
+  $('#angular-my-projects-wrapper').toggle(gallery === Galleries.PRIVATE);
+  $('#public-gallery-wrapper').toggle(gallery === Galleries.PUBLIC);
+}
 
-/**
- * Shows a project at the front of the specified list of published projects.
- * @param {Object} projectData The data for a published project, in the format
- *   defined by projectConstants.projectDataPropType.
- * @param {string} projectType Which list of projects to add this project to.
- *   Valid values include applab, gamelab, playlab, or artist.
- */
-function showNewPublishedProject(projectData, projectType) {
-  getStore().dispatch(prependProjects([projectData], projectType));
+function onShowConfirmPublishDialog(projectId, projectType) {
+  getStore().dispatch(showPublishDialog(projectId, projectType));
 }
 
 // Make this method available to angularProjects.js. This can go away
 // once My Projects is moved to React.
-window.showNewPublishedProject = showNewPublishedProject.bind(this);
+window.onShowConfirmPublishDialog = onShowConfirmPublishDialog.bind(this);
```

Comment:
Looks like this is only used in the form on window. I wonder if we should just do
```
window.onShowConfirmPublishDialog = function(projectId, projectType) {
  getStore().dispatch(...)
}
```